waiter/src/waiter/process_request.clj

```diff
@@ -485,13 +498,22 @@
 
 (defn abort-http-request-callback-factory
   "Creates a callback to abort the http request."
-  [response]
-  (fn abort-http-request-callback [^Exception e]
-    (let [ex (if (instance? IOException e) e (IOException. e))
-          aborted (if-let [request (:request response)]
-                    (.abort request ex)
-                    (log/warn "unable to abort as request not found inside response!"))]
-      (log/info "aborted backend request:" aborted))))
+  [{:keys [abort-ch] :as response}]
+  (fn abort-backend-request-callback [^Exception ex]
+    (let [correlation-id (cid/get-correlation-id)
+          callback (fn abort-http-request-result-callback [aborted?]
+                     (cid/with-correlation-id
+                       correlation-id
+                       (log/info ex "aborted backend request:" aborted?)))]
+      (when-not abort-ch
+        (log/warn "abort-ch not available in response"))
+      (log/debug "aborting backend request")
+      (when-not (and abort-ch (async/put! abort-ch [ex callback]))
+        (log/debug "aborting backend request directly")
+        (if-let [request (:request response)]
```

Comment:
Does the response always have a `:request` key?

Reply:
I expect the response to always have the request key. I added the warn so we are notified when this is not true.